src/optimize.rs

```diff
@@ -57,6 +57,11 @@ pub(crate) fn optimize_instrs_mut(
             instrs.pop();
             instrs.push(Instr::ImplPrim(MemberOfRange, span))
         }
+        // First UnSort = random row
+        ([.., Instr::ImplPrim(UnSort, _)], Instr::Prim(First, span)) => {
```

Comment:
Should do `last` as well.

Reply:
`Instr::Prim(First | Last, span)`

Reply:
I think this would also require changing the error message at algorithm/dyadic/mod.rs:1943

Reply:
I'm not opposed to the error just saying something like "Can't pick random row from empty array."

Reply:
Oh my lord, you guys are replaying my exact conversation with myself from two minutes before I PRd this

Reply:
(I’ll add it)